plugins/SitesManager/angularjs/sites-manager/sites-manager-site.controller.js

```diff
@@ -187,6 +187,25 @@
         };
 
         var deleteSite = function() {
+            var redirectParams = $scope.redirectParams;
+
+            // if the current idSite in the URL is the site we're deleting, then we have to make to change it. otherwise,
+            // if a user goes to another page, the invalid idSite may cause a fatal error.
+            if (broadcast.getValueFromUrl('idSite') === $scope.site.idsite) {
```

Comment:
You need a lose check using `==` instead of `===`. I tested it with Mysqli adapter and didn't work because with Mysqli it is an integer and not a string. Using `==` works.

Reply:
Argh, that's annoying. I think I'll convert the idsite to string, since it's considered bad practice to use `==` in JS (which I'm sure you're aware :) ). (Unless you have an objection.)

Reply:
Where would you cast it to string? It's fine by me but would personally just use `==`. Of course it is a bad practice but it doesn't mean it should never be used or something just for the sake of it. In this case, we know idSite is eg never `0` or something and it doesn't really cause any difference to use `==` vs `===`. Might be easier to implement and read in this case. Up to you though.

Reply:
Yes, but pretty much every linter will complain about that and some transpilers like coffeescript will convert `==` to `===` implicitly. It's really not accepted by the community (not that I'm saying it shouldn't be accepted or anything, it's the opposite in the PHP community), but using it in JS is kind of like adding technical debt, since it will be something that we'd have to change when moving to more modern toolchains. Different languages, different standards.

What I was going to do is just cast `$scope.site.idsite` to string, ie `var currentSite = '' + $scope.site.idsite;` w/ a comment about mysqli.

Reply:
We're not using coffeescript and neither transpile anything etc ;) but totally fine to add the casting as long as it is always updated etc

Reply:
Not yet, but if matomo ever decides to move to Angular 2, we're going to have to start. But I'll just use `==`. If it inspires this much discussion, then I'd rather be on the same page as my teammates then do something that isn't considered a matomo standard :)

Reply:
As said it's all fine to cast ;) When/if we move to angular we will have to fix things everywhere anyway and already cast the output in the API to work with consistent types independent of DB adapter. If it's only a one liner and doesn't break anything else it should be all good to cast.